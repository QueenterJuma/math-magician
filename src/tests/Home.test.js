import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../component/Home.js';

describe('test home component', () => {
  it('should render correctly', () => {
    const homePage = TestRenderer.create(<Home />).toJSON();
    expect(homePage).toMatchSnapshot();
  });
  it('should have a head tag displaying "welcome to our page"', async () => {
    render(<Home />);
    const title = await screen.findByRole('heading');
    expect(title.textContent).toEqual(expect.stringContaining('Welcome to Our Page!'));
  });
});
