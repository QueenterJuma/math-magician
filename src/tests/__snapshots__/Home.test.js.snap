// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home component should render correctly 1`] = `
<div
  className="home"
>
  <h2
    className="heading"
  >
    Welcome to Our Page!
  </h2>
  <p
    className="paragraph"
  >
    Math Magic Calculator is a versatile and powerful tool that can handle wide range of mathematical operations with ease. Its user-friendly interface and intuitive design make it an ideal choice for students, teachers, engineers, and anyone who needs to perform mathematical calculations on a regular basis. The calculator can perform basic arithmetic operations, as well as more complex operations such as logarithms, trigonometric functions, and calculus. It is also flexible in the way it accepts input, allowing users to input expressions in standard notation or through spoken commands. Additionally, it can store and recall previous calculations, making it easy to review and reuse results.
    <br />
    Overall, Math Magic Calculator is an indispensable tool for anyone who needs to work with numbers. Its powerful features and intuitive design make it easy to use and highly efficient.
  </p>
</div>
`;
